contrib/seekable_format/zstd_seekable.h

```diff
@@ -154,21 +155,33 @@ ZSTDLIB_API size_t ZSTD_seekable_writeSeekTable(ZSTD_frameLog* fl, ZSTD_outBuffe
 ZSTDLIB_API ZSTD_seekable* ZSTD_seekable_create(void);
 ZSTDLIB_API size_t ZSTD_seekable_free(ZSTD_seekable* zs);
 
+/*===== Independent seek table management =====*/
+ZSTDLIB_API size_t ZSTD_seekable_copySeekTable(ZSTD_seekable* zs, ZSTD_seekTable** out);
+ZSTDLIB_API size_t ZSTD_seekTable_free(ZSTD_seekTable* st);
+
 /*===== Seekable decompression functions =====*/
 ZSTDLIB_API size_t ZSTD_seekable_initBuff(ZSTD_seekable* zs, const void* src, size_t srcSize);
 ZSTDLIB_API size_t ZSTD_seekable_initFile(ZSTD_seekable* zs, FILE* src);
 ZSTDLIB_API size_t ZSTD_seekable_decompress(ZSTD_seekable* zs, void* dst, size_t dstSize, unsigned long long offset);
 ZSTDLIB_API size_t ZSTD_seekable_decompressFrame(ZSTD_seekable* zs, void* dst, size_t dstSize, unsigned frameIndex);
 
 #define ZSTD_SEEKABLE_FRAMEINDEX_TOOLARGE (0ULL-2)
-/*===== Seek Table access functions =====*/
+/*===== Seekable seek table access functions =====*/
```

Comment:
I only realized that now,
but it seems that the `const` statement is incorrectly positioned.

`ZSTD_seekable* const zs` just means that the pointer `zs` will not be modified, which says nothing about its content, and provides absolutely no value at prototype level.
In contrast, `const ZSTD_seekable* zs` guarantees that the content of `zs` will not be modified, which is very important.

All functions below seem to be read-only. So it would make sense that they all accept a `const ZSTD_seekable* zs` argument. I presume this was simply a mistake, that needs to be fixed.

It impacts your work. Your new API is essentially identical, just applied to `ZSTD_seekTable*` object. Therefore, it inherited the same issue in all the new prototypes.
I believe they should also be updated to accept a `const ZSTD_seekTable* st` argument.